test/application/use.js

```diff
@@ -52,4 +53,11 @@ describe('app.use(fn)', function(){
     .expect(404)
     .end(done);
   });
+
+  it('should throw error for non function', function(done){
+    const app = new Koa();
+
+    should(function(){ app.use('not a function'); }).throw('middleware must be a function!');
```

Comment:
this can be an arrow function now, since we're starting to use them everywhere else